src/fancy-tools/freehandMouseTool.js

```diff
@@ -0,0 +1,1080 @@
+      const currentHandleData = dataHandles[currentHandle];
+      let previousHandleData;
+
+      if (currentHandle === 0) {
+        const lastHandleID = dataHandles.length - 1;
+
+        previousHandleData = dataHandles[lastHandleID];
+      } else {
+        previousHandleData = dataHandles[currentHandle - 1];
+      }
+
+      // Snap back to previous position
+      currentHandleData.x = config.dragOrigin.x;
+      currentHandleData.y = config.dragOrigin.y;
+      previousHandleData.lines[0] = currentHandleData;
+
+      dataHandles.invalidHandlePlacement = false;
+    }
+  }
+
+  /**
+  * Begining of drawing loop when tool is active and a click event happens far
+  * from existing handles.
+  *
+  * @private
+  * @param {Object} eventData - data object associated with an event.
+  */
+  _startDrawing (eventData) {
+    const measurementData = this.createNewMeasurement(eventData);
+    const element = eventData.element;
+    const config = this.configuration;
+
+    // Block event distpatcher whilst drawing
+    mutations.SET_IS_TOOL_LOCKED(true);
+
+    this._referencedElement = element;
+
+    this._activateDraw(element);
+    this._getMouseLocation(eventData);
+
+    addToolState(eventData.element, this.name, measurementData);
+
+    const toolState = getToolState(eventData.element, this.name);
+
+    config.currentTool = toolState.data.length - 1;
+  }
+
+  /**
+  * Adds a point on mouse click in polygon mode.
+  *
+  * @private
+  * @param {Object} eventData - data object associated with an event.
+  */
+  _addPoint (eventData) {
+    const toolState = getToolState(eventData.element, this.name);
+
+    if (toolState === undefined) {
+      return;
+    }
+
+    // Get the toolState from the last-drawn polygon
+    const config = this.configuration;
+    const data = toolState.data[config.currentTool];
+
+    if (data.handles.invalidHandlePlacement) {
+      return;
+    }
+
+    const newHandleData = new FreehandHandleData(eventData.currentPoints.image);
+
+    // If this is not the first handle
+    if (data.handles.length) {
+      // Add the line from the current handle to the new handle
+      data.handles[config.currentHandle - 1].lines.push(eventData.currentPoints.image);
+    }
+
+    // Add the new handle
+    data.handles.push(newHandleData);
+
+    // Increment the current handle value
+    config.currentHandle += 1;
+
+    // Force onImageRendered to fire
+    external.cornerstone.updateImage(eventData.element);
+  }
+
+  /**
+  * If in pencilMode, check the mouse position is farther than the minimum
+  * distance between points, then add a point.
+  *
+  * @private
+  * @param {Object} eventData - Data object associated with an event.
+  * @param {Object} dataHandles - Data object associated with the tool.
+  */
+  _addPointPencilMode (eventData, dataHandles) {
+    const config = this.configuration;
+    const mousePoint = config.mouseLocation.handles.start;
+
+    for (let i = 0; i < dataHandles.length; i++) {
```

Comment:
Might be more clear with Array.some or Array.every? Took me a minute to understand what is supposed to be happening here. You want to ensure that the mouse is farther than 'spacing' away from all points?

if (Array.every(handleFurtherAwayThan(config.spacing)) { this._addPoint }

or 

if (Array.some(handleCloserThanMinimumSpacing)) { return; }